src/machinetalk/messagebus/messagebus.cc

```diff
@@ -266,33 +266,33 @@ static int zmq_setup(msgbusd_self_t *self)
 {
     GOOGLE_PROTOBUF_VERIFY_VERSION;
 
-    // suppress default handling of signals in zctx_new()
+    // suppress default handling of signals in zsock_new()
     // since we're using signalfd()
     zsys_handler_set(NULL);
 
-    self->context = zctx_new ();
+    self->context = zsock_new ();
```

Comment:
I think `zctx` is completely deprecated
`zsock_new` returns a `zsock_t` pointer, not a context pointer

It may be that the `self` struct is wrong now and should not contain a `context` member